tests/test_integration.py

```diff
@@ -37,124 +88,184 @@ def test_vendor_dry_run(python_main_defaults_environment):
 
-# test that solve_environment's DryRunInstall object's
-# 'success' field is True
-def test_solve_environment(python_conda_mirror_main_conda_forge_environment):
+        if version.parse(dep.version) == version.parse("3.9.5"):
+            return
+
+    pytest.fail("python package not found in lock_spec")
 
-    env = Path(str(python_conda_mirror_main_conda_forge_environment))
-    solution = solve_environment(
-        env,
-        "conda",
-        "linux-64",
-    )
 
+# tests that the solve_environment call doesn't fail.  If this doesn't solve
+# a python Exception is raised
+@pytest.mark.integration
+def test_solvable_environment(make_env):
+    env = make_env(environments["conda_mirror"])
+    solution = solve_environment(env, "conda", "linux-64")
 
-def test_notsolvable_environment(python_conda_failing_env):
-    env = Path(str(python_conda_failing_env))
+
+# tests that if the sovle_environment call does fail, an Exception is raised
+@pytest.mark.integration
+def test_notsolvable_environment(make_env):
+    env = make_env(environments["notsolvable"])
     with pytest.raises(Exception):
         solution = solve_environment(env, "conda", "linux-64")
 
 
-# this integration test runs the
-# solver to make sure reconstruct_fetch_actions converts LINK to FETCH
-def test_solve_environment_link2fetch(
-    python_conda_mirror_main_conda_forge_environment,
-):
+# tests that the LINK actions are convert to FETCH actions (happens inside
+# solve environment)
+@pytest.mark.integration
+def test_solve_environment_link2fetch(make_env):
+    expected_packages = {
+        "python": {"version": "3.9.5", "found": False},
+        "conda-mirror": {"version": "0.8.2", "found": False},
+    }
 
-    env = Path(str(python_conda_mirror_main_conda_forge_environment))
+    env = make_env(environments["conda_mirror"])
     package_list = solve_environment(env, "conda", "linux-64")
-    assert any(package["name"] == "python" for package in package_list)
-    assert any(package["version"] == "3.9.5" for package in package_list)
-
-
-# this integration test runs hotfix_vendored_repodata_json
-# which subsequently calls reconstruct_repodata_json
-def test_hotfix_vendored_repodata_json(
-    python_conda_mirror_main_conda_forge_environment, tmp_path_factory
-):
-    env = Path(str(python_conda_mirror_main_conda_forge_environment))
-    temp_dir = tmp_path_factory.mktemp("test-hotfix-repodata")
-    root = create_vendored_dir(
-        "minimal_conda_forge_env", "linux-64", Path(temp_dir)
-    )
+
+    for p in package_list:
+        expected = expected_packages.get(p["name"])
+        if expected is None:
+            continue
+
+        if version.parse(expected["version"]) == version.parse(p["version"]):
+            expected["found"] = True
+
+    for name, val in expected_packages.items():
+        if not val["found"]:
+            pytest.fail("failed to find all the required packages")
+
+
+def _make_directories(tmp_path_factory, environment_name):
```

Comment:
_make_directories could be encapsulated in a pytest class.

Reply:
Why bother?

Reply:
Usually _ in a function name means it is a private function within a class. If this function is used regularly in that one test file, the test file could be reorganized as a pytest class. again, not critical.

Reply:
an _ prefixing a method in python is a gentleman's agreement that it's an implementation detail.  in a class function, that's kind of like private, in a module scoped function it's kind of like "static".  That's what this is an implementation detail and a helper function for the module.  Not for a specific test.

Reply:
![image](https://user-images.githubusercontent.com/117106269/212188291-4c8647a0-d7eb-466e-a93d-c8cf9a4e79dc.png)

That makes sense. And I don't have strong opinions about it. I have just found that if something is needed for more than one test for setup and tear down. The built in setup/fixtures fixtures in a pytest class are helpful. Either way I think what you have done is fine.

Reply:
I hear what you've said.  My preference is to avoid classes until I have a class invariant that I want to maintain.  I think using classes just to group code is an idea from the bad old days of OOP. :).